lib/Partitioner/Partitioner.cpp

```diff
@@ -81,6 +81,117 @@ void Partitioner::initOpMemUsage() {
+
+  // This code assumes all ops are BW limited from SRAM; except
+  // if the input does not fit in SRAM -- then it is DRAM BW limited
+  float peakDramBw = deviceInfo_[0].peakDramBw;
+  float peakSramBw = deviceInfo_[0].peakSramBw;
+  size_t sramCapacity = deviceInfo_[0].sramCapacity;
+  float peakCompute = deviceInfo_[0].peakCompute;
+
+  for (auto &node : F_->getNodes()) {
+    /// compute memory side bytes for inputs from DRAM, SRAM.
+    /// TODO: think about whether this is better off computed inside a Node.
+
+    int n = node.getNumInputs();
+    uint64_t sizeDram = 0;
+    uint64_t sizeSram = 0;
+    if (node.getKind() == Kinded::Kind::SaveNodeKind) {
+      computeTime_[&node] = 0.0f;
+      continue;
+    }
+
+    /// The memory bytes for embedding table lookups is data dependent,
+    /// so it needs to be calculated as per the number of indices accessed.
+    if (node.getKind() == Kinded::Kind::SparseLengthsWeightedSumNodeKind) {
+      auto *SLWSN = llvm::dyn_cast<SparseLengthsWeightedSumNode>(&node);
+      /// compute how many entries of the embedding table we look up
+      auto numLookups = SLWSN->getIndices().getNode()->dims(0).front();
+      /// compute how many bytes we read per lookup
+      auto tableSize = SLWSN->getData().getNode()->getType(0)->getSizeInBytes();
+      auto numRows = SLWSN->getData().getNode()->dims(0).front();
+      auto sizePerLookup = tableSize / numRows;
+      /// compute total bytes read
+      uint64_t sizeInput = numLookups * sizePerLookup;
+
+      /// does the table fit in SRAM or DRAM
+      if (tableSize > sramCapacity) {
+        sizeDram += sizeInput;
+      } else {
+        sizeSram += sizeInput;
+      }
+
+      /// we also read the indices, weights and lengths arrays
+      sizeSram += SLWSN->getIndices().getNode()->getType(0)->getSizeInBytes();
+      sizeSram += SLWSN->getWeights().getNode()->getType(0)->getSizeInBytes();
+      sizeSram += SLWSN->getLengths().getNode()->getType(0)->getSizeInBytes();
+    } else {
+      /// for all other ops, iterate through all inputs and get size in bytes
+      for (int i = 0; i < n; i++) {
+        auto ty = node.getNthInput(i).getNode()->getType(0);
+        uint64_t sizeInput = ty->getSizeInBytes();
+        if (sizeInput > sramCapacity) {
+          sizeDram += sizeInput;
+        } else {
+          sizeSram += sizeInput;
+        }
+      }
+    }
+
+    // Repeat for outputs
+    if (node.getNumResults() > 0) {
+      auto myty = node.getType(0);
+      uint64_t sizeOutput = myty->getSizeInBytes();
+      if (sizeOutput > sramCapacity) {
+        sizeDram += sizeOutput;
+      } else {
+        sizeSram += sizeOutput;
+      }
+    }
+
+    /// Calculate compute ops. Currently only computed for Matmul, Conv, FC
+    /// TODO: think about whether this is better off computed inside a Node.
+    uint64_t totalOps = 0;
+    if (node.getKind() == Kinded::Kind::MatMulNodeKind) {
+      auto *MMN = llvm::dyn_cast<MatMulNode>(&node);
```

Comment:
I prefer using "switch". If we need to add more node type here, "switch" looks better:)

Reply:
Makes sense.